js_modules/dagit/packages/core/src/asset-graph/AssetNode.tsx

```diff
@@ -318,46 +345,126 @@ export const ComputeStatusNotice: React.FC<{computeStatus: AssetComputeStatus}>
 
 export const AssetLatestRunWithNotices: React.FC<{
   liveData?: LiveDataForNode;
-}> = ({liveData}) => {
+  includeFreshness?: boolean;
+}> = ({liveData, includeFreshness}) => {
   const {
     lastMaterialization,
+    lastMaterializationRunStatus,
     unstartedRunIds,
     inProgressRunIds,
     runWhichFailedToMaterialize,
     stepKey,
   } = liveData || MISSING_LIVE_DATA;
 
-  return inProgressRunIds?.length > 0 ? (
-    <Box flex={{gap: 4, alignItems: 'center'}}>
-      <Tooltip content="A run is currently rematerializing this asset.">
-        <Spinner purpose="body-text" />
-      </Tooltip>
-      <AssetRunLink runId={inProgressRunIds[0]} />
-    </Box>
-  ) : unstartedRunIds?.length > 0 ? (
-    <Box flex={{gap: 4, alignItems: 'center'}}>
-      <Tooltip content="A run has started that will rematerialize this asset soon.">
-        <Spinner purpose="body-text" stopped />
-      </Tooltip>
-      <AssetRunLink runId={unstartedRunIds[0]} />
-    </Box>
-  ) : runWhichFailedToMaterialize?.__typename === 'Run' ? (
-    <Box flex={{gap: 4, alignItems: 'center'}}>
-      <Tooltip
-        content={`Run ${titleForRun({
-          runId: runWhichFailedToMaterialize.id,
-        })} failed to materialize this asset`}
-      >
-        <Icon name="warning" color={Colors.Red500} />
-      </Tooltip>
-      <AssetRunLink runId={runWhichFailedToMaterialize.id} />
+  const runTagContent =
+    inProgressRunIds?.length > 0 ? (
+      <Box flex={{gap: 4, alignItems: 'center'}}>
+        <Tooltip content="A run is currently rematerializing this asset.">
+          <Spinner purpose="caption-text" />
+        </Tooltip>
+        <AssetRunLink runId={inProgressRunIds[0]} />
+      </Box>
+    ) : unstartedRunIds?.length > 0 ? (
+      <Box flex={{gap: 4, alignItems: 'center'}}>
+        <Tooltip content="A run has started that will rematerialize this asset soon.">
+          <Spinner purpose="caption-text" stopped />
+        </Tooltip>
+        <AssetRunLink runId={unstartedRunIds[0]} />
+      </Box>
+    ) : runWhichFailedToMaterialize?.__typename === 'Run' ? (
+      <Box flex={{gap: 4, alignItems: 'center'}}>
+        <Tooltip
+          content={`Run ${titleForRun({
+            runId: runWhichFailedToMaterialize.id,
+          })} failed to materialize this asset`}
+        >
+          <Icon name="warning" color={Colors.Red500} />
+        </Tooltip>
+        <AssetRunLink runId={runWhichFailedToMaterialize.id} />
+      </Box>
+    ) : lastMaterialization ? (
+      <Box flex={{gap: 6, alignItems: 'center'}}>
+        {lastMaterializationRunStatus ? (
+          <RunStatusIndicator status={lastMaterializationRunStatus} size={10} />
+        ) : (
+          <div style={{width: 10}} />
+        )}
+        <AssetRunLink
+          runId={lastMaterialization.runId}
+          event={{stepKey, timestamp: lastMaterialization.timestamp}}
+        />
+      </Box>
+    ) : undefined;
```

Comment:
I may have commented on this elsewhere, but I find these nested ternaries hard to follow. :(

Reply:
Ahh will change this @hellendag! I need to find another pattern for doing this sort of thing, I guess a sub-render function with a series of if/returns? I think this one could be a whole sub-component maybe but it might take a bunch of props.